src/extension/src/scripts/generate-test.ts

```diff
@@ -0,0 +1,119 @@
+import { CoreTemplateStudio } from "../coreTemplateStudio";
+import { CONSTANTS } from "../constants";
+
+let instance: CoreTemplateStudio;
+let backends: string[] = [];
+let frontends: string[] = [];
+let pagesObj: any[] = [];
+const projType = "FullStackWebApp";
+let syncAttemptNum = 0;
+let prevPromise: Promise<any> = Promise.resolve(null);
+
+const delay = (time: number) => {
+  return new Promise(function(resolve) {
+    setTimeout(resolve, time);
+  });
+};
+
+let getPagesObj = (
+  instance: CoreTemplateStudio,
+  frontend: string,
+  backend: string
+) => {
+  return instance
+    .getPages(projType, frontend, backend)
+    .then(pages => {
+      pagesObj = [];
+      pages.forEach((page: { name: any; templateId: any }) => {
```

Comment:
we can simplify this even more using:
```ts
return pages.map((page: { name: string; templateId: string }) => {
  return {name: page.name, identity: page.templateId};
}
```

Eliminating the need for pagesObj.

Reply:
good call! I just made the changes